src/languageservice/services/yamlLinks.ts

```diff
@@ -5,19 +5,25 @@
 import { findLinks as JSONFindLinks } from 'vscode-json-languageservice/lib/umd/services/jsonLinks';
 import { DocumentLink } from 'vscode-languageserver';
 import { TextDocument } from 'vscode-languageserver-textdocument';
+import { Telemetry } from '../../languageserver/telemetry';
 import { yamlDocumentsCache } from '../parser/yaml-documents';
 
-export function findLinks(document: TextDocument): Promise<DocumentLink[]> {
-  try {
-    const doc = yamlDocumentsCache.getYamlDocument(document);
-    // Find links across all YAML Documents then report them back once finished
-    const linkPromises = [];
-    for (const yamlDoc of doc.documents) {
-      linkPromises.push(JSONFindLinks(document, yamlDoc));
+export class YamlLinks {
+  constructor(private readonly telemetry: Telemetry) {}
+
+  findLinks(document: TextDocument): Promise<DocumentLink[]> {
+    try {
+      const doc = yamlDocumentsCache.getYamlDocument(document);
+      // Find links across all YAML Documents then report them back once finished
+      const linkPromises = [];
+      for (const yamlDoc of doc.documents) {
+        linkPromises.push(JSONFindLinks(document, yamlDoc));
+      }
+      // Wait for all the promises to return and then flatten them into one DocumentLink array
+      return Promise.all(linkPromises).then((yamlLinkArray) => [].concat(...yamlLinkArray));
+    } catch (err) {
+      console.warn(err.toString());
```

Comment:
Do we need this console.warn here or was that just for debugging?

Reply:
Oh, that is for debugging, I forgot to remove it